mlserver/parallel/worker.py

```diff
@@ -132,22 +150,38 @@ async def _process_model_update(self, update: ModelUpdateMessage):
                 "Unknown model update message with type ", update.update_type
             )
 
+    def _update_cb(self, update_task: Task):
+        err = update_task.exception()
+        if err:
+            logger.error(e)
+
+        self._model_updates.task_done()
+
+    def send_request(self, request_message: InferenceRequestMessage):
+        """
+        Send an inference request message to the worker.
+        Note that this method should be both multiprocess- and thread-safe.
+        """
+        loop = asyncio.get_event_loop()
+        self._requests.put(request_message)
+
     async def send_update(self, model_update: ModelUpdateMessage):
         """
         Send a model update to the worker.
         Note that this method should be both multiprocess- and thread-safe.
         """
         loop = asyncio.get_event_loop()
-        self.model_updates.put(model_update)
-        await loop.run_in_executor(self._executor, self.model_updates.join)
+        self._model_updates.put(model_update)
+        await loop.run_in_executor(self._executor, self._model_updates.join)
```

Comment:
do we still need `send_update` to be `async`?

Reply:
Yes, because model update messages can be awaited until the worker confirms that the model has been loaded / unloaded.